src/topology/algebra/ring.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2018 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot, Johannes Hölzl
+
+Theory of topological rings.
+-/
+
+import topology.algebra.group ring_theory.ideals
+
+open classical set lattice filter topological_space
+local attribute [instance] classical.prop_decidable
+
+section topological_ring
+universes u v w
+variables (α : Type u) [topological_space α]
+
+/-- A topological semiring is a semiring where addition and multiplication are continuous. -/
+class topological_semiring [semiring α]
+  extends topological_add_monoid α, topological_monoid α : Prop
+
+variables [ring α]
+
+/-- A topological ring is a ring where the ring operations are continuous. -/
+class topological_ring extends topological_add_monoid α, topological_monoid α : Prop :=
+(continuous_neg : continuous (λa:α, -a))
+
+variables [t : topological_ring α]
+instance topological_ring.to_topological_semiring : topological_semiring α := {..t}
+
+instance topological_ring.to_topological_add_group : topological_add_group α := {..t}
+end topological_ring
+
+section topological_comm_ring
+variables {α : Type*} [topological_space α] [comm_ring α] [topological_ring α]
+
+def ideal.closure (S : ideal α) : ideal α :=
+{ carrier := closure S,
+  zero := subset_closure S.zero_mem,
+  add  := assume x y hx hy,
+    mem_closure2 continuous_add' hx hy $ assume a b, S.add_mem,
+  smul  := assume c x hx,
+    have continuous (λx:α, c * x) := continuous_mul continuous_const continuous_id,
+    mem_closure this hx $ assume a, S.mul_mem_left }
+
+@[simp] lemma ideal.coe_closure (S : ideal α) :
+  (S.closure : set α) = closure S := rfl
+
+end topological_comm_ring
+
+section ideal_is_add_subgroup
+variables {α : Type*} [comm_ring α] {M : Type*} [add_comm_group M] [module α M] (N : submodule α M)
+
+instance submodule_is_add_subgroup : is_add_subgroup ↑N :=
+{ zero_mem := N.zero,
+  add_mem  := N.add,
+  neg_mem  := λ _, N.neg_mem }
+
+end ideal_is_add_subgroup
+
+section topological_ring
+variables {α : Type*} [topological_space α] [comm_ring α] [topological_ring α] (N : ideal α)
+open ideal.quotient
+
+instance topological_ring_quotient_topology : topological_space N.quotient :=
+by dunfold ideal.quotient submodule.quotient; apply_instance
+
+-- this should be quotient_add_saturate ...
```

Comment:
Isn't it a good occasion to rename the lemma?

Reply:
I removed the comment, the current name is clearly better than the suggestion.